packages/xdl/src/Analytics.ts

```diff
@@ -9,59 +9,59 @@ const PLATFORM_TO_ANALYTICS_PLATFORM: { [platform: string]: string } = {
   linux: 'Linux',
 };
 
-let _userId: string | undefined;
-let _userTraits: any;
-
 export class AnalyticsClient {
-  private segmentNodeInstance: Segment | undefined;
-  private version: string | undefined;
-  private userIdentifyCalled: boolean = false;
+  private _userId: string | undefined;
+  private _userTraits: any;
+  private _segmentNodeInstance: Segment | undefined;
+  private _version: string | undefined;
+
+  public get version() {
+    return this._version;
+  }
+
+  public get userId() {
+    return this._userId;
+  }
 
   public flush() {
-    if (this.segmentNodeInstance) {
-      this.segmentNodeInstance.flush();
+    if (this._segmentNodeInstance) {
+      this._segmentNodeInstance.flush();
     }
   }
 
   public setSegmentNodeKey(key: string) {
     // Do not wait before flushing, we want node to close immediately if the programs ends
-    this.segmentNodeInstance = new Segment(key, { flushInterval: 300 });
+    this._segmentNodeInstance = new Segment(key, { flushInterval: 300 });
   }
 
-  public setUserProperties(userId: string, traits: any) {
-    _userId = userId;
-    _userTraits = traits;
+  public identifyUser(userId: string, traits: any) {
+    this._userId = userId;
+    this._userTraits = traits;
 
-    this.ensureUserIdentified();
+    if (this._segmentNodeInstance) {
+      this._segmentNodeInstance.identify({
+        userId: this._userId,
+        traits: this._userTraits,
+        context: this.getContext(),
+      });
+    }
   }
 
   public setVersionName(version: string) {
```

Comment:
if you have public getter and setter then just make the field public and remove both better and setter

Reply:
I removed the setter and moved the setting logic into an `initializeClient` method and kept the getter.